components/data_stores/actions/add-update-multiple-records/add-update-multiple-records.mjs

```diff
@@ -23,82 +24,81 @@ export default {
+      }
+
+      // Tries to convert string to possible other primitive types
+      if (typeof value === "string") {
+        // Convert boolean
+        if (value.toLowerCase() === "true") {
+          return true;
+        }
+        if (value.toLowerCase() === "false") {
+          return false;
+        }
+
+        // Convert number
+        if (!isNaN(value)) {
+          return parseFloat(value);
+        }
+      }
+
+      // Try to evaluate string as javascript, using xss as extra security
+      // If some problem occurs, return the original string
+      try {
+        return eval(`(${xss(value)})`);
+      } catch {
+        return value;
+      }
     },
-    //Handling all serialized and unserialized inputs
-    //to treat values as objects no matter how they entered.
-    _parseData(obj) {
-      if (typeof obj === "string") {
-        const sanitizedValue = this.data_stores.sanitizeJson(obj);
-        const json = JSON.parse(sanitizedValue);
-        return this._parseDataRecursive(json);
-      } else if (typeof obj === "object") {
-        const newObj = {};
-        for (const [
-          key,
-          value,
-        ] of Object.entries(obj)) {
-          if (typeof value === "string") {
-            try {
-              const sanitizedValue = this.data_stores.sanitizeJson(value);
-              const json = JSON.parse(sanitizedValue);
-              if (typeof json !== "object") {
-                newObj[key] = json;
-              } else {
-                newObj[key] = this._parseDataRecursive(json);
-              }
-            } catch (err) {
-              newObj[key] = value;
-            }
-          } else if (typeof value === "object") {
-            newObj[key] = this._parseDataRecursive(value);
-          } else {
-            newObj[key] = value;
-          }
+    /**
+     * Add all the key-value pairs in the map object to be used in the data store
+     * @param {*} data Data to be parsed
+     * @param {Map} map Hashmap to be updated
+     */
+    populateHashMapOfData(data, map) {
+      if (!Array.isArray(data) && typeof(data) === "object") {
+        const keys = Object.keys(data);
+        for (const key of keys) {
+          map[key] = this.convertString(data[key]);
+        }
+        return;
+      }
+
+      if (Array.isArray(data)) {
+        for (const item of data) {
+          this.populateHashMapOfData(item, map);
         }
-        return newObj;
       }
-      throw new Error("Unsupported expression");
+    },
+    getHashMapOfData(data) {
+      const map = {};
+      this.populateHashMapOfData(data, map);
+      return map;
     },
   },
   async run({ $ }) {
     if (typeof this.data === "string") {
-      try {
-        const sanitizedValue = this.data_stores.sanitizeJson(this.data);
-        JSON.parse(sanitizedValue);
-      } catch (err) {
-        throw new Error("please provide an object or object array!");
-      }
+      this.data = eval(`(${this.data})`);
```

Comment:
Should this use `xss` too?

```suggestion
      this.data = eval(`(${xss(this.data}))`);
```

Reply:
No need, it will be sanitized later if needed on `convertString`